keps/sig-node/5328-node-declared-features/README.md

```diff
@@ -0,0 +1,1013 @@
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- All functionality completed
+- All security enforcement completed
+- All monitoring requirements completed
+- All testing requirements completed
+- All known pre-release issues and gaps resolved
+
+**Note:** Beta criteria must include all functional, security, monitoring, and testing requirements along with resolving all issues and gaps identified
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- Allowing time for feedback
+- All issues and gaps identified as feedback during beta are resolved
+
+**Note:** GA criteria must not include any functional, security, monitoring, or testing requirements.  Those must be beta requirements.
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+**Alpha:**
+*   The `NodeDeclaredFeatures` feature gate is implemented and disabled by default.
+*   The Kubelet correctly discovers and declares features to the `node.status.declaredFeatures` list when the feature gate is enabled.
+*   A shared library is implemented with the initial logic for feature inference and matching.
+*   The API server correctly serves and validates the `declaredFeatures` field when the feature gate is enabled.
+*   The `kube-scheduler` and `admission-controller` plugins are introduced and integrated with the shared library.
+*   All unit and integration tests outlined in the Test Plan are implemented and verified.
+
+**Beta:**
+*   Cluser Autoscaler support for existing nodepools with active nodes. 
```

Comment:
Any reasons this is not included in alpha?

Reply:
Ack. Will keep this as a stretch-goal and will update the KEP if we can get it done in 1.35.